torch/lib/TensorLib/CUDAGenerator.cpp

```diff
@@ -0,0 +1,44 @@
+#ifdef TENSORLIB_CUDA_ENABLED
```

Comment:
Can we make these uniform by making a THState and consistently passing it everywhere in normal TH? But I guess applying this consistently would involve upheaving a lot of code, and increasing register pressure on these functions calls for not a very good reason. So maybe not.